site/public/js/ta-grading.js

```diff
@@ -108,86 +100,128 @@ $(function () {
     }
   }
 
-  $('#settings-popup').on('change', '.ta-grading-setting-option', function() {
-    var storageCode = $(this).attr('data-storage-code');
-    if(storageCode) {
-      localStorage.setItem(storageCode, this.value);
-      if(settingsCallbacks && settingsCallbacks.hasOwnProperty(storageCode)) {
-        settingsCallbacks[storageCode](this.value);
+  if (typeof TA_GRADING_DO_NOT_LOAD_PANELS === 'undefined' || !TA_GRADING_DO_NOT_LOAD_PANELS) {
+    Object.assign(taLayoutDet, getSavedTaLayoutDetails());
+    // Check initially if its the mobile screen view or not
+    isMobileView = window.innerWidth <= MOBILE_WIDTH;
+    initializeTaLayout();
+
+    window.addEventListener('resize', () => {
+      let wasMobileView = isMobileView;
+      isMobileView = window.innerWidth <= MOBILE_WIDTH;
+      // if the width is switched between smaller and bigger screens, re-initialize the layout
+      if (wasMobileView !== isMobileView) {
+        initializeTaLayout();
       }
-    }
-  })
+    });
+
+    $('#settings-popup').on('change', '.ta-grading-setting-option', function() {
+      var storageCode = $(this).attr('data-storage-code');
+      if(storageCode) {
+        localStorage.setItem(storageCode, this.value);
+        if(settingsCallbacks && settingsCallbacks.hasOwnProperty(storageCode)) {
+          settingsCallbacks[storageCode](this.value);
+        }
+      }
+    })
 
-  // Progress bar value
-  let value = $(".progressbar").val() ? $(".progressbar").val() : 0;
-  $(".progress-value").html("<b>" + value + '%</b>');
+    // Progress bar value
+    let value = $(".progressbar").val() ? $(".progressbar").val() : 0;
+    $(".progress-value").html("<b>" + value + '%</b>');
```

Comment:
Would be cleaner to write this as `<b>${value} %</b>` (surrounded by backticks--I can't get them to show up in MD).  I'm surprised that eslint didn't catch this.  Usually it does.